### PR TITLE
OKTA-685809: Added external link support in sidenav

### PR DESCRIPTION
**Pull request by ishan-krishna-okta**
## Description:
- **What's changed?** Added external link support in sidenav
- **Is this PR related to a Monolith release?** No

### Resolves:

* [OKTA-685809](https://oktainc.atlassian.net/browse/OKTA-685809)

### CODE REVIEW

File: packages/@okta/vuepress-theme-prose/components/SidebarItem.vue [hunk 1]
```diff
@@ -6,7 +6,7 @@
       hidden: hidden }"
   >
     <router-link
-      v-if="entityType === types.link"
+      v-if="entityType === types.link && link.path && !link.path.toLowerCase().startsWith('http') && !link.path.toLowerCase().startsWith('www.')"
```
Comment (emikhasyak-okta) [inline packages/@okta/vuepress-theme-prose/components/SidebarItem.vue:9]: entitiyType is dynamically calculated, let's leverage that functionality.
We can add smartLink type here https://github.com/okta/okta-developer-docs/blob/master/packages/%40okta/vuepress-theme-prose/components/SidebarItem.vue#L121
And we can add checks for smartLink here https://github.com/okta/okta-developer-docs/blob/master/packages/%40okta/vuepress-theme-prose/components/SidebarItem.vue#L128
All entity type calculations will be in the same place which easier to maintain.

On the component level, we will have simple checks like `v-if="entityType === types.link"` and `v-if="entityType === types.smartLink"`

Comment (ishan-krishna-okta) [reply packages/@okta/vuepress-theme-prose/components/SidebarItem.vue]: Excellent point. 

I have updated the PR with the suggested changes.

**Review by vanngo-okta — APPROVED** [review]
LGTM

**Review by emikhasyak-okta — APPROVED** [review]
Great job! 🚢

**Pull request merged**